src/modules/visitor.rs

```diff
@@ -0,0 +1,105 @@
+use syntax::ast;
+use syntax::parse::token::{DelimToken, Token};
+use syntax::parse::{stream_to_parser_with_base_dir, Directory, ParseSess};
+use syntax::symbol::kw;
+use syntax::visit::Visitor;
+use syntax_pos::Symbol;
+
+pub(crate) struct ModItem {
+    pub(crate) item: ast::Item,
+}
+
+/// Traverse `cfg_if!` macro and fetch modules.
+pub(crate) struct CfgIfVisitor<'a> {
+    parse_sess: &'a ParseSess,
+    mods: Vec<ModItem>,
+    base_dir: Directory<'a>,
+}
+
+impl<'a> CfgIfVisitor<'a> {
+    pub(crate) fn new(parse_sess: &'a ParseSess, base_dir: Directory<'a>) -> CfgIfVisitor<'a> {
+        CfgIfVisitor {
+            mods: vec![],
+            parse_sess,
+            base_dir,
+        }
+    }
+
+    pub(crate) fn mods(self) -> Vec<ModItem> {
+        self.mods
+    }
+}
+
+impl<'a, 'ast: 'a> Visitor<'ast> for CfgIfVisitor<'a> {
+    fn visit_mac(&mut self, mac: &'ast ast::Mac) {
+        match self.visit_mac_inner(mac) {
+            Ok(()) => (),
+            Err(e) => debug!("{}", e),
```

Comment:
should the error be propagated?

Reply:
The error means the `cfg_if!` macro is ill-formed. Nothing we can do here, so rustfmt should just ignore this.